File: Converge/Models/ConvergeSettings.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

using System;
using System.Collections.Generic;

namespace Converge.Models
{
    public class ConvergeSettings
    {
        public List<string> PreferredBuildings { get; set; }

        public string ZipCode { get; set; }

        public GPSCoordinates GeoCoordinates { get; set; }

        public bool IsConvergeUser { get; set; }

        public List<string> MyList { get; set; }

        public List<string> LikedSections { get; set; }

        public List<string> DislikedSections { get; set; }

        public DateTime? LastNPSDate { get; set; }

        /// <summary>
        /// This is a list of Yelp IDs.
        /// </summary>
        public List<string> FavoriteVenuesToCollaborate { get; set; }

        /// <summary>
        /// This is a list of identities that correspond to workspaces or conference rooms.
        /// </summary>
        public List<string> FavoriteCampusesToCollaborate { get; set; }

        public DateTime? ConvergeInstalledDate { get; set; }

        public List<string> RecentBuildingUpns { get; set; }
    }
}

File: Converge/ClientApp/src/types/ConvergeSettings.ts
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { GeoCoordinates } from "@microsoft/microsoft-graph-types";

interface ConvergeSettings {
  preferredBuildings?: string[];
  zipCode?: string;
  isConvergeUser?: boolean;
  myList?: string[];
  likedSections?: string[];
  dislikedSections?: string[];
  lastNPSDate?: string;
  favoriteVenuesToCollaborate?: string[];
  favoriteCampusesToCollaborate?: string[];
  convergeInstalledDate?: string;
  recentBuildingUpns?: string[];
  geoCoordinates?: GeoCoordinates;
}

export default ConvergeSettings;
